src/IdentityServer4/ResponseHandling/TokenResponseGenerator.cs

```diff
@@ -227,6 +229,68 @@ protected virtual Task<TokenResponse> ProcessPasswordRequestAsync(TokenRequestVa
             };
         }
 
+        /// <summary>
+        /// Processes the response for device code grant request.
+        /// </summary>
+        /// <param name="request">The request.</param>
+        /// <returns></returns>
+        protected virtual async Task<TokenResponse> ProcessDeviceCodeRequestAsync(TokenRequestValidationResult request)
+        {
+            Logger.LogTrace("Creating response for device code request");
+
+            //////////////////////////
+            // access token
+            /////////////////////////
+            var (accessToken, refreshToken) = await CreateAccessTokenAsync(request.ValidatedRequest);
```

Comment:
I was looking through the PRs and noticed `.ConfigureAwait(false)` isn't used in the codebase... But I wonder if I comment here in some PR and ask would it be OK to introduce them little-by-little? :)

Reply:
Good point. There isn't one in ASP.NET Core and it appears they aren't writing that out, see at https://blog.stephencleary.com/2017/03/aspnetcore-synchronization-context.html for more authoritative explanation. (That article recommends it for libraries, though.)

This brings a question, how much of this code could be run elsewhere? As for an example, some [Orleans](https://github.com/dotnet/orleans/) people are experimenting with IS. The system works so that it has roughly one threas per core and it gives "single-thread guarantee" (https://dotnet.github.io/orleans/Documentation/Advanced-Concepts/External-Tasks-and-Grains.html), which could be problematic. (`Task.Run` can be used to excape blocking behaviour if needed, though.)

That being written, I don't know if this has been a concious decision as in to save in trouble of writing it everywhere or if it might avoid some other trouble. So I thought just to ask it in some context. :)

Reply:
Okay, I'd have to defer to Dom & Brock about support throughout the rest of the library From what I've seen `configureawait` is only necessary when targetting .NET Framework. 

I'm going to assume that this PR won't be the place to start any change in behavior though. Maybe you could open a separate issue?